### PR TITLE
Add parameter to increment seed on MonteCarloAbsorbtion

### LINKED ISSUE #29017: use different random # sequences for diff detectors in MonteCarloAbsorption
At the moment, if you run a MonteCarloAbsorption calculation on a workspace with multiple spectra (multiple detectors), the same random number sequence gets used for each thread. This results in the same set of scattering points being used for each thread eg run the following script and note that the scatter point stats reported in debug mode are the same for each detector:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

# load example data (TOF)
data = CreateSampleWorkspace(WorkspaceType='Histogram', Function='Powder Diffraction',BinWidth=10)
data = ConvertUnits(data, Target="Wavelength")
# Create cylindrical sample in (hollow) cylindrical can          
SetSample(data,
        Geometry={'Shape': 'Cylinder', 'Height': 4.0,
                  'Radius': 2.0, 'Center': [0.,0.,0.]},
        Material={'ChemicalFormula': 'Ni', 'MassDensity': 7.0},
        ContainerGeometry={'Shape': 'HollowCylinder', 'Height': 4.0,
                  'InnerRadius': 2.0, 'OuterRadius': 2.3,
                  'Center': [0.,0.,0.]},
        ContainerMaterial={'ChemicalFormula': 'Al'})

# create rectangular beam profile 0.8cm x 1cm
SetBeam(data, Geometry={'Shape': 'Slit', 'Width': 0.8, 'Height': 1.0})

abscor_sampleandcontainer = MonteCarloAbsorption(InputWorkspace=data, EventsPerPoint=1000)
# apply the correction
corrected = data/abscor_sampleandcontainer
```
It would be better if the results for each detector were independent and used their own random number sequence. This would remove any bias in the random number sequences used. It would also improve the behaviour of any spatial interpolation carried out if the errors on the simulated points were independent.
This can be achieved by moving the following declaration of the random number generator outside of the parallel for loop in MonteCarloAbsorption.cpp:

MersenneTwister rng(seed);

### PR DESCRIPTION
**Pull request by Harrietbrown**
**Description of work.**

This PR adds an optional parameter to the MonteCarloAbsorbtion algorithm that will cause a different seed to be used for each detector when set to true. previously the rng would use the same seed for each detector.

**To test:**

Run the script from the issue, first as is, and then with `IncrementSeed=True` in `MonteCarloAbsorption`. the output should be similar but not exactly the same. test that it is no slower that without incrementing the seed.

When running with the same inital seed it should have the same outcome.

Fixes #29017

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by DannyHindson — CHANGES_REQUESTED** [review]
Hi Harriet, I've had a look at this and I don't think we need the additional "increment seed" parameter. I can't think of a situation where it's beneficial to use the same random number sequence across all detectors. This is unfortunately going to mean that the reference files for various system tests will need changing because the new functionality will be "on" in all cases.
As a side note - if we had kept the new parameter, we would have had to have added it to the end of the parameter list to ensure backwards compatibility for any users running MonteCarloAbsorption using positional parameters

**Review by DannyHindson — APPROVED** [review]
Tested it using the following script which is where I first noticed the problem:
```
# import mantid algorithms, numpy and matplotlib
from mantid.simpleapi import *
import matplotlib.pyplot as plt
import numpy as np

# load example data (TOF)
data = CreateSampleWorkspace(WorkspaceType='Histogram', Function='Powder Diffraction',BinWidth=10)
data = ConvertUnits(data, Target="Wavelength")
# Create cylindrical sample in (hollow) cylindrical can          
SetSample(data,
        Geometry={'Shape': 'Cylinder', 'Height': 4.0,
                  'Radius': 2.0, 'Center': [0.,0.,0.]},
        Material={'ChemicalFormula': 'Ni', 'MassDensity': 7.0},
        ContainerGeometry={'Shape': 'HollowCylinder', 'Height': 4.0,
                  'InnerRadius': 2.0, 'OuterRadius': 2.3,
                  'Center': [0.,0.,0.]},
        ContainerMaterial={'ChemicalFormula': 'Al'})

# create rectangular beam profile 0.8cm x 1cm
SetBeam(data, Geometry={'Shape': 'Slit', 'Width': 0.8, 'Height': 1.0})

abscor_sampleandcontainer = MonteCarloAbsorption(InputWorkspace=data, EventsPerPoint=1000)
# apply the correction
corrected = data/abscor_sampleandcontainer
```
The scatter points stats (counts of where the simulation generates a scatter point in sample vs container) are now slightly different for each detector so looks good.

Also did the following tests which both worked:
- that you get the same results on different runs with same random number seed
- that changing the random number seed changes the results

I had wondered if there would be further system tests affected eg complexShapesAbsorptionsTest. I had a quick look at this one and it only runs the absorption correction for a single detector. So the fact it doesn't change is correct.

So in summary all looks good thanks
